### PR TITLE
[FEATURE] Implemented UI for adding / deleting and updating variants

### PR DESCRIPTION
**Pull request by turbo-ele**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | https://github.com/sulu/SuluProductBundle/issues/315
| Related issues/PRs | none
| License | MIT

#### What's in this PR?

This PR contains the UI implementation for adding, deleting and updating product variants.

### CODE REVIEW

File: Api/Product.php [hunk 1]
```diff
@@ -1298,6 +1299,22 @@ public function getTags()
...
      * @VirtualProperty
+     * @SerializedName("variantAttributes")
+     *
+     * @return Attribute[]
+     */
+    public function getVariantAttributes()
+    {
+        $variantAttributes = [];
+        foreach ($this->entity->getVariantAttributes() as $attribute) {
+            $variantAttributes[] = new Attribute($attribute, $this->locale, $this->locale);
```
Comment (janmassive) [inline Api/Product.php]: $this->fallbackLocale

File: Resources/public/js/components/products/components/variants-list/main.js [hunk 2]
```diff
@@ -8,227 +8,280 @@
...
+            var deferred = $.Deferred();
+
+            var currencies = new Currencies({locale: this.options.locale});
+            currencies.fetch({
+                success: function(data) {
+                    this.currencies = data.toJSON();
+                    deferred.resolve();
+                }.bind(this),
+                error: function() {
+                    console.error('Error while loading currencies');
```
Comment (janmassive) [inline Resources/public/js/components/products/components/variants-list/main.js]: Write directly to console?

File: Resources/public/js/components/products/components/variants-list/main.js [hunk 3]
```diff
@@ -8,227 +8,280 @@
...
+                this.$overlay = this.sandbox.dom.createElement('<div>');
+                this.sandbox.dom.append(this.$el, this.$overlay);
+
+                // Create content.
+                this.sandbox.start([{
+                    name: 'variant-overlay@suluproduct',
+                    options: {
+                        el: this.$overlay,
+                        data: data,
+                        currencies: this.currencies,
```
Comment (janmassive) [inline Resources/public/js/components/products/components/variants-list/main.js]: Provide in function?

**Comment by janmassive** [comment]
Else ack

File: Resources/public/js/components/variant-overlay/main.js [hunk 4]
```diff
@@ -0,0 +1,584 @@
...
+         * Called when warnings CANCEL button is clicked.
+         *
+         * @returns {boolean}
+         */
+        onWarningCancelClicked = function() {
+            this.sandbox.emit(retrieveOverlayEventName.call(this, 'slide-left'));
+
+            // Set back local to previous selection.
+            var locales = this.sandbox.sulu.locales;
+            this.sandbox.emit('husky.select.null.update', locales , [locales.indexOf(this.options.locale)], false);
```
Comment (turbo-ele) [inline Resources/public/js/components/variant-overlay/main.js]: yes

File: Resources/public/js/models/variant.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+                // Handle parameters.
+                if (this.ids && this.ids.length > 0) {
+                    params.ids = this.ids.join(',');
+                }
+                if (this.locale) {
+                    params.locale = this.locale;
+                }
+                if (Object.keys(params).length) {
+                    url += '?' + $.param(params);
```
Comment (turbo-ele) [inline Resources/public/js/models/variant.js]: yes

File: Resources/public/js/components/variant-overlay/main.js [hunk 6]
```diff
@@ -0,0 +1,584 @@
...
+        /**
+         * Constructor of component.
+         */
+        initialize: function() {
+            this.changeLocale = null;
+            this.options = this.sandbox.util.extend(true, {}, defaults, this.options);
+
+            // Render overlay.
+            render.call(this);
+
```
Comment (turbo-ele) [inline Resources/public/js/components/variant-overlay/main.js]: done

**Pull request merged**